src/main/java/service/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,22 @@
+package service.operation;
+
+import dao.FruitDao;
+import dao.FruitDaoImpl;
+
+public class PurchaseOperation implements OperationHandler {
+    private FruitDao fruitDao;
+
+    public PurchaseOperation() {
+        this.fruitDao = new FruitDaoImpl();
+    }
+
+    @Override
+    public void handle(String fruit, int quantity) {
+        int balanceAfterPurchase = fruitDao.getFruitBalance(fruit) - quantity;
+        if (balanceAfterPurchase >= 0) {
+            fruitDao.updateBalanceOfFruit(fruit,balanceAfterPurchase);
```

Comment:
The method name `updateBalanceOfFruit` suggests that it might be updating the balance to any arbitrary value. It's better to have a method that clearly indicates that the balance is being decreased due to a purchase, such as `decreaseFruitBalance`. This enhances the readability and maintainability of the code.